src/with-render.js

```diff
@@ -10,7 +10,7 @@ function attachShadow (elem) {
 
 export const withRender = (Base = HTMLElement) => class extends Base {
   propsChangedCallback () {
-    this[_shadowRoot] = this[_shadowRoot] || (this[_shadowRoot] = attachShadow(this));
+    this[_shadowRoot] = this[_shadowRoot] || (this[_shadowRoot] = (this.shadowRoot || attachShadow(this)));
```

Comment:
If they call `attachShadow({ mode: 'closed' })` this would fail, right? I think the only reliable way to do this would be to override `attachShadow`, but I'm not sure we should go down that route.